lib/matplotlib/axes/_axes.py

```diff
@@ -4995,13 +5002,37 @@ def fill_betweenx(self, y, x1, x2=0, where=None,
                 continue
 
             N = len(yslice)
-            Y = np.zeros((2 * N + 2, 2), float)
+            Y = np.zeros((2 * N + 2, 2), np.float)
+            if interpolate:
+                def get_interp_point(ind):
```

Comment:
scipy isn't a dependency, so using their functions isn't an option unless you copy paste them in a module of ours.